htlcswitch/hop/iterator.go

```diff
@@ -113,6 +114,108 @@ func (r *sphinxHopIterator) ExtractErrorEncrypter(
 	return extracter(r.ogPacket.EphemeralKey)
 }
 
+// BlindingProcessor is an interface that provides the cryptographic operations
+// required for processing blinded hops.
+type BlindingProcessor interface {
+	// DecryptBlindedHopData decrypts a blinded blob of data using the
+	// ephemeral key provided.
+	DecryptBlindedHopData(ephemPub *btcec.PublicKey,
+		encryptedData []byte) ([]byte, error)
+}
+
+// BlindingKit contains the components required to extract forwarding
+// information for hops in a blinded route.
+type BlindingKit struct {
+	// BlindingPoint holds a blinding point that was passed to the node via
+	// update_add_htlc's TLVs.
```

Comment:
For the `BlindingKit` this is only the blinding point from `update_add_htlc` because it's what we pass into payload parsing to help decrypt the onion - ie, it's only used if we're a relaying node in the blinded path. 

When we pull the blinding point out of the payload, we go directly to setting it in our `ForwardingInfo` (covered in the next two commits).